include/gatekeeper_config.h

```diff
@@ -25,66 +25,18 @@
 #ifndef _GATEKEEPER_CONFIG_H_
 #define _GATEKEEPER_CONFIG_H_
 
-/* 
- * XXX #14 Sample parameters for test only.
- * They should be configured in the configuration step.
- */
-#define GATEKEEPER_MAX_PORTS	(4)
-#define GATEKEEPER_MAX_QUEUES	(8)
-
-/*
- * XXX #14 Sample parameters for test only.
- * They should be analyzed or tested further to find optimal values.
- *
- * Larger queue size can mitigate bursty behavior, but can also increase 
- * pressure on cache and lead to lower performance.
- */
-#define GATEKEEPER_NUM_RX_DESC	(128)
-#define GATEKEEPER_NUM_TX_DESC	(512)
-
-/* 
- * XXX #14 Sample parameter for the number of elements in the mbuf pool.
- * This should be analyzed or tested further to find optimal value.
- *
- * The optimum size (in terms of memory usage) for a mempool is when it is a 
- * power of two minus one.
- *
- * Need to provision enough memory for the worst case,
- * since each queue needs at least
- * GATEKEEPER_NUM_RX_DESC + GATEKEEPER_NUM_TX_DESC + max_pkt_burst
- * descriptors. i.e., GATEKEEPER_DESC_PER_QUEUE =
- * (GATEKEEPER_NUM_RX_DESC + GATEKEEPER_NUM_TX_DESC \
- *		+ max_pkt_burst (let's say 32)) = 672.
- *
- * So, the pool size should be at least the maximum number of queues * 
- *		number of descriptors per queue, i.e., 
- * (GATEKEEPER_MAX_PORTS * GATEKEEPER_MAX_QUEUES * \
- *              GATEKEEPER_DESC_PER_QUEUE - 1) = 5376.
- */
-#define GATEKEEPER_MBUF_SIZE (8191)
-
-/* 
- * XXX #14 Sample parameter for the size of the per-core object cache,
- * i.e., number of struct rte_mbuf elements in the per-core object cache.
- * this should be analyzed or tested further to find optimal value.
- *
- * Each core deals with at most GATEKEEPER_MAX_PORTS queues, so the cache size
- * should be at least (number of ports * number of descriptors per queue), i.e.,
- * (GATEKEEPER_MAX_PORTS * GATEKEEPER_DESC_PER_QUEUE).
- * 
- * Notice that, this argument must be lower or equal to 
- * CONFIG_RTE_MEMPOOL_CACHE_MAX_SIZE and n / 1.5. 
- * It is advised to choose cache_size to have "n modulo cache_size == 0": 
- * if this is not the case, some elements will always stay in the pool 
- * and will never be used. Here, n is GATEKEEPER_MBUF_SIZE.
- *
- * The maximum cache size can be adjusted in DPDK's .config file: 
- * CONFIG_RTE_MEMPOOL_CACHE_MAX_SIZE.
- */
-#define GATEKEEPER_CACHE_SIZE	(512)
-
 extern const uint16_t LUA_MSG_MAX_LEN;
 
+/* Configuration for the Gatekeeper functional blocks. */
+struct gatekeeper_config {
+	uint8_t  gatekeeper_max_ports;
```

Comment:
1. Instead of defining another global `struct` in Gatekeeper, shouldn't we move the fields of `struct gatekeeper_config` into `struct net_config`? The fields in `struct gatekeeper_config` are even related to the network.

2. The constants that this patch is moving to the configuration files were created before we have the current structure of configuration files. So, isn't it possible for at least some of these constants to be dynamically calculated as we do with lcores and number of queues for example?

Reply:
2. It doesn't seem to be the case. For example, `GATEKEEPER_MAX_PORTS` is used in `gatekeeper_init_network()` only to check that the number of ports doesn't go above that limit. So, if we remove that constant from Gatekeeper, Gatekeeper should still work fine. The real need for `GATEKEEPER_MAX_PORTS` is to find `GATEKEEPER_MBUF_SIZE` and `GATEKEEPER_CACHE_SIZE` as their comments point out. Moreover, the number of needed ports can be easily computed from variables `front_ports` and `back_ports` in `lua/net.lua`.

Although we need to work further on patch **config: make net compile-time parameters runtime parameters**, all the other patches in this pull request are ready for merge. So I suggest to move patch **config: make net compile-time parameters runtime parameters** into a new pull request. Once you do this move, remember to edit the description of the current pull request to only close issue #57 since the new pull request will close issue #14.